File: app/src/main/java/com/cleanup/todoc/domaine/usecase/GetTasksUseCase.java
package com.cleanup.todoc.domaine.usecase;

import androidx.lifecycle.LiveData;

import com.cleanup.todoc.datasource.model.TaskEntity;
import com.cleanup.todoc.data.repository.TaskRepository;
import com.cleanup.todoc.domaine.model.TaskDomain;

import java.util.List;

public class GetTasksUseCase {
    private final TaskRepository repository;
    public GetTasksUseCase(TaskRepository repository) {
        this.repository = repository;
    }
    public LiveData<List<TaskDomain>> getAllTasks() {
        return repository.getAllTasks();
    }
}

File: app/src/main/java/com/cleanup/todoc/domaine/usecase/GetTasksUseCase.kt
package com.cleanup.todoc.domaine.usecase

import com.cleanup.todoc.data.repository.TaskRepository
import com.cleanup.todoc.domaine.TaskDomainUiMapper
import com.cleanup.todoc.domaine.model.TaskDomain
import com.cleanup.todoc.presentation.model.Task
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map

class GetTasksUseCase(private val repository: TaskRepository) {
    val allTasks: Flow<List<Task>>
        get() = repository.allTasks.map { tasks: List<TaskDomain> ->
            TaskDomainUiMapper.mapToTaskList(
                tasks
            )
        }
}